src/pages/abs/[id]/abstract.tsx

```diff
@@ -4,33 +4,102 @@ import { abstractPageNavDefaultQueryFields } from '@components/AbstractSideNav/m
 import { GetServerSideProps, NextPage } from 'next';
 import Head from 'next/head';
 import { isNil } from 'ramda';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { normalizeURLParams } from 'src/utils';
 import Link from 'next/link';
 import Image from 'next/image';
 import { createUrlByType } from '@components/AbstractSources/linkGenerator';
+import clsx from 'clsx';
+import { AbsLayout } from '@components/Layout/AbsLayout';
 
 export interface IAbstractPageProps {
   doc?: IDocsEntity;
   error?: Error;
+  userData: IUserData;
+  params: {
+    q: string;
+    fl: string[];
+    sort: SolrSort[];
+  };
 }
 
+const MAX_AUTHORS = 20;
+
 const AbstractPage: NextPage<IAbstractPageProps> = (props: IAbstractPageProps) => {
-  const { doc, error } = props;
+  const { doc, error, userData, params } = props;
+
+  const [showNumAuthors, setShowNumAuthors] = useState<number>(MAX_AUTHORS);
+
+  const [aff, setAff] = useState({ show: false, data: [] as string[] });
+
+  // onComponentDidMount
+  useEffect(() => {
+    if (showNumAuthors > doc.author.length) {
+      setShowNumAuthors(doc.author.length);
+    }
+  }, []);
+
+  const handleShowAllAuthors = () => {
+    setShowNumAuthors(doc.author.length);
+  };
+
+  const handleShowLessAuthors = () => {
+    setShowNumAuthors(Math.min(doc.author.length, MAX_AUTHORS));
+  };
+
+  const handleShowAff = () => {
+    const adsapi = new AdsApi({ token: userData.access_token });
```

Comment:
You should switch to use the hook probably
```suggestion
    const { api } = useAPI();
```

Reply:
Updated using suggested